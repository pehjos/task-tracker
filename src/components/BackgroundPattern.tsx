import { FC } from "react";

export const BackgroundPattern: FC = () => {
  return (
    <div className="fixed inset-0 z-0 overflow-hidden pointer-events-none" data-testid="background-pattern">
      {/* Grid Pattern */}
      <div className="absolute inset-0 opacity-10" data-testid="grid-pattern">
        <div className="h-full w-full bg-[linear-gradient(to_right,#6366f1_1px,transparent_1px),linear-gradient(to_bottom,#6366f1_1px,transparent_1px)] bg-[size:60px_60px]" />
      </div>

      {/* Accent Shapes */}
      <div className="absolute -top-16 -left-16 w-80 h-80 bg-blue-400 rounded-full mix-blend-multiply filter blur-2xl opacity-10" data-testid="accent-shape" />
      <div className="absolute top-10 -right-16 w-80 h-80 bg-purple-400 rounded-full mix-blend-multiply filter blur-2xl opacity-10 delay-1500" data-testid="accent-shape" />
      <div className="absolute bottom-0 left-1/3 w-80 h-80 bg-indigo-400 rounded-full mix-blend-multiply filter blur-2xl opacity-10 delay-3000" data-testid="accent-shape" />

      {/* Icons */}
      <div className="absolute inset-0 opacity-50 flex items-center justify-center">
        <div className="absolute top-20 -left-2 md:left-0 lg:left-20 text-blue-400 text-6xl transform rotate-12 opacity-60 delay-1000" data-testid="icon">
          &#x2714;
        </div>
        <div className="absolute bottom-20 -left-5 md:left-0 lg:left-20 text-blue-400 text-6xl transform rotate-12 opacity-60 delay-1000" data-testid="icon">
          &#x1F4DD;
        </div>
        <div className="absolute bottom-32 -right-4 md:-right-5 lg:right-20 text-purple-400 text-6xl transform -rotate-6 opacity-60 delay-2000" data-testid="icon">
          &#x231B;
        </div>
        <div className="absolute top-40 -right-2 md:-right-2 lg:right-10 text-indigo-400 text-6xl transform rotate-3 opacity-60 delay-3000" data-testid="icon">
          &#x1F4C4;
        </div>
      </div>
    </div>
  );
};
