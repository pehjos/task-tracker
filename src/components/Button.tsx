import { FC, MouseEvent ,JSX} from "react";

interface ButtonProps {
  label: string;
  type: "button" | "submit" | "reset";
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void; 
  icon?: JSX.Element;
  styles?: string;
  arialLabel?: string;
  isDropdown?: boolean;
}

export const Button: FC<ButtonProps> = ({ label, type, onClick, icon, styles, arialLabel, isDropdown }) => {
  return (
    <button
      className={styles}
      onClick={(event) => onClick?.(event)}
      aria-label={arialLabel ?? label}
      type={type}
    >
      {!isDropdown && <div className="absolute inset-0 bg-white/20 rounded-xl opacity-0 group-hover:opacity-100 transition-opacity duration-200" />}
      <div className={`flex items-center ${!isDropdown ? "space-x-2" : ""}`}>
        {icon && icon}
        <span className="font-medium">{label}</span>
      </div>
    </button>
  );
};
