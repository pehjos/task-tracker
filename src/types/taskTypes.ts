export type Priority = "Low" | "Medium" | "High";

export interface Task {
  id: number; 
  title: string; 
  description: string; 
  priority: Priority;
  dueDate: string; 
}

export interface TaskContextType {
  tasks: Task[];
  displayedTasks: Task[];
  addTask: (task: Omit<Task, "id">) => void;
  editTask: (task: Task) => void;
  deleteTask: (id: number) => void;
  handleSearch: (searchTerm: string) => void;
  handleFilter: (priority: Priority | "all") => void;
  isLoading: boolean;
  showForm: boolean;
  toggleForm: (open?: boolean) => void;
  setShowForm: React.Dispatch<React.SetStateAction<boolean>>;
  taskToEdit: Task | null;
  setTaskToEdit: (task: Task | null) => void;
}

