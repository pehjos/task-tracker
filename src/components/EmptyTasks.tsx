import { FC } from "react";
import { FaClipboardList } from "react-icons/fa";
import { HiSparkles } from "react-icons/hi";
import { Button } from "./Button";

interface EmptyTasksProps {
  toggleForm: () => void; 
}

export const EmptyTasks: FC<EmptyTasksProps> = ({ toggleForm }) => {
  return (
    <div className="relative overflow-hidden">
      <div className="relative flex flex-col items-center justify-center p-12 rounded-2xl border border-gray-100 bg-white/70 backdrop-blur-lg min-h-[400px] space-y-8">
        <div className="flex flex-col items-center space-y-6 max-w-md">
          <div className="relative">
            <div className="absolute inset-0 bg-blue-200 rounded-full blur-md animate-pulse" />
            <div className="relative bg-gradient-to-tr from-blue-500 to-purple-500 p-5 rounded-full">
            <FaClipboardList className="w-14 h-14 text-white" data-testid="task-icon" />

            </div>
            <div className="absolute -top-2 -right-2">
            <HiSparkles className="w-6 h-6 text-yellow-400 animate-bounce" data-testid="sparkles-icon" />
            </div>
          </div>

          <div className="text-center space-y-3">
            <h3 className="text-2xl font-bold bg-gradient-to-r from-blue-600 to-purple-600 bg-clip-text text-transparent">
              Your Task List Awaits
            </h3>
            <p className="text-gray-600 leading-relaxed">
              Turn your ideas into tasks and get things done.
            </p>
          </div>

          <Button
            type="button"
            onClick={toggleForm}
            label="Create First Task"
            styles="group relative px-6 py-3 bg-gradient-to-r from-blue-600 to-purple-600 text-white rounded-xl hover:shadow-lg hover:shadow-blue-500/25 transition-all duration-200 transform hover:-translate-y-0.5"
          />
        </div>

        {/* Dots (Decorative Elements) */}
        <div className="absolute top-8 right-8 grid grid-cols-2 gap-1">
          {[...Array(4)].map((_, i) => (
            <div key={i} className="w-2 h-2 rounded-full bg-gray-200" />
          ))}
        </div>
        <div className="absolute bottom-8 left-8 grid grid-cols-2 gap-1">
          {[...Array(4)].map((_, i) => (
            <div key={i} className="w-2 h-2 rounded-full bg-gray-200" />
          ))}
        </div>
      </div>
    </div>
  );
};
