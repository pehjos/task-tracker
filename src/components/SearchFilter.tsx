import { FC, useState, useEffect } from "react";
import { BiSearch } from "react-icons/bi";
import { Button } from "./Button";
import { Priority } from "../types/taskTypes";

interface SearchFilterProps {
  handleSearch: (searchTerm: string) => void;
  handleFilter: (filter: Priority | "all") => void;
}

export const SearchFilter: FC<SearchFilterProps> = ({ handleSearch, handleFilter }) => {
  const [searchTerm, setSearchTerm] = useState("");
  const [activeFilter, setActiveFilter] = useState<Priority | "all">("all");

  useEffect((): (() => void) => {
    const delayDebounce = setTimeout(() => {
      handleSearch(searchTerm);
    }, 500); 
  
    return (): void => clearTimeout(delayDebounce);
  }, [searchTerm]); 

  const filterTasks = (filter: Priority | "all"): void => {
    setActiveFilter(filter);
    handleFilter(filter);
  };

  return (
    <div className="flex flex-col md:flex-row gap-4 mt-6">
      <form onSubmit={(e) => e.preventDefault()} className="relative flex-grow">
        <Button
          type="submit"
          label=""
          arialLabel="Submit button"
          onClick={undefined}
          icon={<BiSearch className="h-5 w-5 text-gray-400" />}
          styles="absolute inset-y-0 left-0 flex items-center pl-3 cursor-pointer"
        />
        <input
          data-testid="search-input"
          type="text"
          placeholder="Search tasks..."
          className="w-full pl-10 pr-4 py-2.5 bg-gray-50 border border-gray-200 rounded-xl text-sm md:text-base text-gray-700 placeholder-gray-400 transition-all outline-none duration-200 focus:outline-none focus:border-blue-300 focus:ring-1 focus:ring-blue-300"
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
      </form>

      {/* Filter Options */}
      <div className="flex justify-between md:justify-start gap-2 md:gap-3 p-1 overflow-x-auto hideScrollbar">
        <FilterCheckbox label="All" filterValue="all" activeFilter={activeFilter} onChange={filterTasks} />
        <FilterCheckbox label="Low" filterValue="Low" activeFilter={activeFilter} onChange={filterTasks} />
        <FilterCheckbox label="Medium" filterValue="Medium" activeFilter={activeFilter} onChange={filterTasks} />
        <FilterCheckbox label="High" filterValue="High" activeFilter={activeFilter} onChange={filterTasks} />
      </div>
    </div>
  );
};

interface FilterCheckboxProps {
  label: string;
  filterValue: Priority | "all";
  activeFilter: Priority | "all";
  onChange: (filter: Priority | "all") => void;
}

const FilterCheckbox: FC<FilterCheckboxProps> = ({ label, filterValue, activeFilter, onChange }) => (
  <label className={`shrink-0 flex items-center gap-2 px-3 sm:px-4 py-2 rounded-lg text-sm font-medium transition-all duration-200 cursor-pointer ${activeFilter === filterValue ? "bg-blue-100 text-blue-700 shadow-sm" : "text-gray-600 hover:bg-gray-100"}`}>
    <input
      type="checkbox"
      checked={activeFilter === filterValue}
      onChange={() => onChange(filterValue)}
      className="h-4 w-4 text-blue-600 rounded focus:ring-0 transition duration-150 cursor-pointer"
    />
    {label}
  </label>
);
