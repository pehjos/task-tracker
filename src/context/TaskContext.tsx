import { createContext, useContext, useState, useEffect, useMemo, ReactNode, JSX } from "react";
import { getTasksFromLocalStorage, saveTasksToLocalStorage } from "../utils/localStorage";
import { Task, Priority,TaskContextType } from "../types/taskTypes";

const TaskContext = createContext<TaskContextType | null>(null);

// Provider Component
export const TaskProvider = ({ children }: { children: ReactNode }): JSX.Element => {
  const [tasks, setTasks] = useState<Task[]>([]);
  const [displayedTasks, setDisplayedTasks] = useState<Task[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const [showForm, setShowForm] = useState(false);
  const [taskToEdit, setTaskToEdit] = useState<Task | null>(null);

  const [searchTerm, setSearchTerm] = useState<string>("");
const [priorityFilter, setPriorityFilter] = useState<Priority | "all">("all"); 


  // Sort tasks by ID
  const sortedTasks = useMemo(() => tasks.slice().sort((a, b) => b.id - a.id), [tasks]);

  const updateTasks = (updatedTasks: Task[]): void => {
    const sortedUpdatedTasks = updatedTasks.slice().sort((a, b) => b.id - a.id); 
    setTasks(sortedUpdatedTasks);
    setDisplayedTasks(sortedUpdatedTasks);
  };

  const addTask = (task: Omit<Task, "id">): void => {
    if (task.title.trim()) {
      const newTask: Task = { ...task, id: Date.now() };
      updateTasks([...tasks, newTask]);
    }
  };

  const editTask = (updatedTask: Task): void => {
    updateTasks(tasks.map((task) => (task.id === updatedTask.id ? updatedTask : task)));
  };

  const deleteTask = (id: number): void => {
    updateTasks(tasks.filter((task) => task.id !== id));
  };
  const updateDisplayedTasks = (): void => {
    let filteredTasks = tasks;
  
    
    if (priorityFilter !== "all") {
      filteredTasks = filteredTasks.filter((task) => task.priority === priorityFilter);
    }
  
  
    if (searchTerm.trim()) {
      filteredTasks = filteredTasks.filter((task) => 
        task.title.toLowerCase().includes(searchTerm.trim().toLowerCase())
      );
    }
  
    setDisplayedTasks(filteredTasks); 
  };
  
  const handleSearch = (term: string): void => {
    setSearchTerm(term); 
  };
  
  const handleFilter = (priority: Priority | "all"): void => {
    setPriorityFilter(priority); 
  };
  
  
  

  const toggleForm = (open?: boolean): void => {
    setShowForm(open !== undefined ? open : !showForm);
  };
  useEffect(() => {
    updateDisplayedTasks();
  }, [searchTerm, priorityFilter, tasks]);
  

  useEffect(() => {
    const storedTasks = getTasksFromLocalStorage();
    if (storedTasks && storedTasks.length > 0) {
      setTasks(storedTasks);
      setDisplayedTasks(storedTasks);
    }
    setIsLoading(false);
  }, []);

  useEffect(() => {
    if (!isLoading) {
      saveTasksToLocalStorage(tasks);
    }
  }, [tasks, isLoading]);

  return (
    <TaskContext.Provider
      value={{
        tasks: sortedTasks,
        displayedTasks,
        addTask,
        editTask,
        deleteTask,
        handleSearch,
        handleFilter,
        isLoading,
        showForm,
        toggleForm,
        taskToEdit,
        setTaskToEdit,
        setShowForm
      }}
    >
      {children}
    </TaskContext.Provider>
  );
};

// Custom Hook to Use Context
export const useTasks = (): TaskContextType => {
  const context = useContext(TaskContext);
  if (!context) {
    throw new Error("useTasks must be used within a TaskProvider");
  }
  return context;
};
