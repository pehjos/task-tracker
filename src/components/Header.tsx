import { FC } from "react";
import { BiPlusCircle } from "react-icons/bi";
import { Button } from "./Button";
import { SearchFilter } from "./SearchFilter";
import { Priority } from "../types/taskTypes";

interface HeaderProps {
  toggleForm: () => void;
  handleSearch: (search: string) => void;
  handleFilter: (filter: Priority | "all") => void;
}

export const Header: FC<HeaderProps> = ({ toggleForm, handleSearch, handleFilter }) => {
  return (
    <header className="sticky top-0 z-50 px-3  bg-white pb-3 ">
      <div className="relative  my-3">
        <div className="flex flex-wrap items-center justify-between gap-4">
          <div className="relative group">
            <h1 className="text-left text-2xl md:text-3xl font-extrabold tracking-tight">
              <span className="bg-gradient-to-r from-blue-600 via-purple-600 to-blue-600 bg-[length:200%_auto] animate-gradient bg-clip-text text-transparent">
                My Tracker
              </span>
            </h1>
            <p className="mt-1 text-sm md:text-base text-gray-600 hidden sm:block transition-opacity duration-200">
              Stay organized and productive
            </p>
          </div>

          <Button
            type="button"
            label="New Task"
            onClick={toggleForm}
            icon={<BiPlusCircle className="w-5 h-5" />}
            styles="relative px-5 md:px-6 py-2.5 md:py-3 bg-gradient-to-r from-blue-600 to-purple-600 text-white rounded-xl font-medium text-sm md:text-base transition-all duration-20 hover:shadow-lg hover:shadow-blue-500/20 active:scale-[0.98] overflow-hidden group"
          />
        </div>

        {/* Ensure Correct Filter Values */}
        <SearchFilter handleFilter={(filter) => handleFilter(filter as Priority | "all")} handleSearch={handleSearch} />
      </div>
    </header>
  );
};
