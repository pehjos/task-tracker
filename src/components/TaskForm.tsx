import { useEffect, useState, FC } from "react";
import { useTasks } from "../context/TaskContext"; 
import { Task } from "../types/taskTypes";
import { Button } from "./Button";




interface TaskFormProps {
  addTask: (task: Omit<Task, "id">) => void;
  editTask: (task: Task) => void;
  taskToEdit: Task | null;
  toggleForm: (open?: boolean) => void;
  setTaskToEdit: (task: Task | null) => void;
}

export const TaskForm: FC<TaskFormProps> =() => {
  const { addTask, editTask, toggleForm, taskToEdit, setTaskToEdit } = useTasks(); // âœ… Use context
  const [task, setTask] = useState<Task>({
    id: Date.now(), // Temporary ID
    title: "",
    description: "",
    priority: "Low",
    dueDate: "", 
  });

  const [errors, setErrors] = useState<{ title?: string; dueDate?: string }>({});

  useEffect(() => {
    if (taskToEdit) {
      setTask(taskToEdit);
    } else {
      setTask({
        id: Date.now(),
        title: "",
        description: "",
        priority: "Low",
        dueDate: "",
      });
    }
  }, [taskToEdit]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>): void => {
    const { name, value } = e.target;
    setTask((prevTask) => ({
      ...prevTask,
      [name]: value,
    }));
  };

  const validateForm = (): boolean => {
    const errors: { title?: string; dueDate?: string } = {};
  
    if (!task.title.trim()) errors.title = "Title is required";
    if (!task.dueDate) errors.dueDate = "Due date is required";
  
    setErrors(errors);
    return Object.keys(errors).length === 0;
  };
  
  const closeForm = (): void => {
    toggleForm(false); 
    setTask({ id: Date.now(), title: "", description: "", priority: "Low", dueDate: "" }); 
    setTaskToEdit(null); 
    setErrors({});
  };

  const handleSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    if (!validateForm()) return;

    if (taskToEdit) {
      editTask({ ...taskToEdit, ...task });
    } else {
      addTask(task);
    }

    closeForm();
  };

  return (
    <div className="fixed inset-0 bg-black/30 backdrop-blur-sm bg-opacity-60 flex items-center justify-center z-50">
      <div className="w-full max-w-lg p-6 bg-white rounded-xl shadow-lg transform transition-all duration-300 sm:max-w-xl">
        <h2 className="text-xl font-semibold mb-6 text-center text-gray-800">
          {taskToEdit ? "Edit Task" : "Add New Task"}
        </h2>
        <form className="space-y-5" onSubmit={handleSubmit}>
          {/* Task Title */}
          <div>
            <label htmlFor="title" className="block text-sm font-medium text-gray-600 text-left">Title*</label>
            <input  id="title" name="title"
              type="text"
              value={task.title}
              onChange={handleChange}
              className="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
              placeholder="Enter task title"
              required
            />
            {errors.title && <p className="text-sm text-red-500">{errors.title}</p>}
          </div>

          {/* Task Description */}
          <div>
            <label className="block text-sm font-medium text-gray-600 text-left">Description</label>
            <textarea
              name="description"
              value={task.description}
              onChange={handleChange}
              rows={3}
              className="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
              placeholder="Enter task description"
            />
          </div>

          {/* Task Priority */}
          <div>
            <label className="block text-sm font-medium text-gray-600 text-left">Priority</label>
            <select
              name="priority"
              value={task.priority}
              onChange={handleChange}
              className="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
            >
              <option value="Low">Low</option>
              <option value="Medium">Medium</option>
              <option value="High">High</option>
            </select>
          </div>

          {/* Task Due Date  */}
          <div>
            <label htmlFor="dueDate" className="block text-sm font-medium text-gray-600 text-left">Due Date *</label>
            <input
              type="date"
              name="dueDate"
              id="dueDate"
              value={task.dueDate}
              onChange={handleChange}
              className="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
              required
            />
            {errors.dueDate && <p className="text-sm text-red-500">{errors.dueDate}</p>}
          </div>

          {/* Buttons */}
          <div className="flex justify-end gap-3">
            <Button
              type="button"
              label="Cancel"
               data-testid="cancel-button"
              onClick={closeForm}
              styles="px-6 py-2 bg-white text-gray-700 rounded-md shadow-sm hover:bg-gray-100 border border-gray-300 transition duration-150 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-400 transform hover:-translate-y-0.5"
            />
            <Button
           
              type="submit"
              label={taskToEdit ? "Save Changes" : "Add Task"}
              onClick={undefined}
              styles="px-4 py-2 bg-gradient-to-r from-blue-600 to-purple-600 text-white rounded-md shadow-md hover:shadow-lg hover:shadow-purple-400/50 transition duration-150 transform hover:-translate-y-0.5 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500"
            />
          </div>
        </form>
      </div>
    </div>
  );
};
