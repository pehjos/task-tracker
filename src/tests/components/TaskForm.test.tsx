import { render, screen, fireEvent } from "@testing-library/react";
import { TaskForm } from "../../components/TaskForm";
import { describe, it, expect, vi } from "vitest";
import { Task } from "../../types/taskTypes";
import { TaskProvider } from "../../context/TaskContext"; 

const renderWithProvider = (taskToEdit: Task | null = null): ReturnType<typeof render> => {
  const mockAddTask = vi.fn();
  const mockEditTask = vi.fn();
  const mockToggleForm = vi.fn();
  const mockSetTaskToEdit = vi.fn();

  return render(
    <TaskProvider>
      <TaskForm
        addTask={mockAddTask}
        editTask={mockEditTask}
        toggleForm={mockToggleForm}
        setTaskToEdit={mockSetTaskToEdit}
        taskToEdit={taskToEdit}
      />
    </TaskProvider>
  );
};

describe("TaskForm Component", () => {
  it("renders correctly", () => {
    renderWithProvider();
    expect(screen.getByText(/Add New Task/i)).toBeInTheDocument();
  });

  it("closes the form when cancel is clicked", () => {
    renderWithProvider();
    

  });

  it("calls `addTask` when submitting a new task", () => {
    const { container } = renderWithProvider();
    
    fireEvent.change(screen.getByPlaceholderText(/enter task title/i), { target: { value: "New Task" } });
    fireEvent.change(screen.getByPlaceholderText(/enter task description/i), { target: { value: "Task Description" } });
    expect(container).toMatchSnapshot();
  });

  
});
