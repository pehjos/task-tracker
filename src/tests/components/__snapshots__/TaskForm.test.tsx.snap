// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskForm Component > calls \`addTask\` when submitting a new task 1`] = `
<div>
  <div
    class="fixed inset-0 bg-black bg-opacity-60 flex items-center justify-center z-50"
  >
    <div
      class="w-full max-w-lg p-6 bg-white rounded-xl shadow-lg transform transition-all duration-300 sm:max-w-xl"
    >
      <h2
        class="text-xl font-semibold mb-6 text-center text-gray-800"
      >
        Add New Task
      </h2>
      <form
        class="space-y-5"
      >
        <div>
          <label
            class="block text-sm font-medium text-gray-600 text-left"
            for="title"
          >
            Title
          </label>
          <input
            class="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
            id="title"
            name="title"
            placeholder="Enter task title"
            required=""
            type="text"
            value="New Task"
          />
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-600 text-left"
          >
            Description
          </label>
          <textarea
            class="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
            name="description"
            placeholder="Enter task description"
            rows="3"
          >
            Task Description
          </textarea>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-600 text-left"
          >
            Priority
          </label>
          <select
            class="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
            name="priority"
          >
            <option
              value="Low"
            >
              Low
            </option>
            <option
              value="Medium"
            >
              Medium
            </option>
            <option
              value="High"
            >
              High
            </option>
          </select>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-600 text-left"
            for="dueDate"
          >
            Due Date
          </label>
          <input
            class="w-full px-4 py-2 mt-1 bg-gray-100 rounded-md shadow-inner border border-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-400 focus:border-transparent transition duration-150"
            id="dueDate"
            name="dueDate"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="flex justify-end gap-3"
        >
          <button
            aria-label="Cancel"
            class="px-6 py-2 bg-white text-gray-700 rounded-md shadow-sm hover:bg-gray-100 border border-gray-300 transition duration-150 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-400 transform hover:-translate-y-0.5"
            type="button"
          >
            <div
              class="absolute inset-0 bg-white/20 rounded-xl opacity-0 group-hover:opacity-100 transition-opacity duration-200"
            />
            <div
              class="flex items-center space-x-2"
            >
              <span
                class="font-medium"
              >
                Cancel
              </span>
            </div>
          </button>
          <button
            aria-label="Add Task"
            class="px-4 py-2 bg-gradient-to-r from-blue-600 to-purple-600 text-white rounded-md shadow-md hover:shadow-lg hover:shadow-purple-400/50 transition duration-150 transform hover:-translate-y-0.5 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500"
            type="submit"
          >
            <div
              class="absolute inset-0 bg-white/20 rounded-xl opacity-0 group-hover:opacity-100 transition-opacity duration-200"
            />
            <div
              class="flex items-center space-x-2"
            >
              <span
                class="font-medium"
              >
                Add Task
              </span>
            </div>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
